web/web.go

```diff
@@ -240,9 +241,16 @@ func MustBuildServerURL(pathPrefix string) string {
 	if err != nil {
 		panic(err)
 	}
-	hostname, err := os.Hostname()
+	hostname, err := getServerHostname()
 	if err != nil {
 		panic(err)
 	}
 	return fmt.Sprintf("http://%s:%s%s", hostname, port, pathPrefix)
 }
+
+func getServerHostname() (string, error) {
+	if *hostname != "" {
+		return *hostname, nil
+	}
+	return os.Hostname()
+}
```

Comment:
Please configure your editor to save text files with a newline character after the last line. This prevents all kinds of issues, like this warning here, noise in diffs, as well as http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline.